tradeit-android-sdk/src/main/java/it/trade/android/sdk/manager/TradeItLinkedBrokerManager.java

```diff
@@ -64,27 +64,23 @@ public TradeItLinkedBrokerManager(TradeItApiClientParcelable apiClient, TradeItL
     public synchronized void syncLocalLinkedBrokers(List<TradeItLinkedBrokerData> linkedBrokerDataList) throws TradeItSaveLinkedLoginException, TradeItDeleteLinkedLoginException {
         List<TradeItLinkedBrokerParcelable> linkedBrokers = this.linkedBrokers;
 
-        // Add missing linkedBrokers
-        for (TradeItLinkedBrokerData linkedBrokerData: linkedBrokerDataList) {
-            TradeItLinkedBrokerParcelable linkedBrokerParcelable = createNewLinkedBroker(linkedBrokerData);
-            TradeItLinkedLoginParcelable linkedLoginParcelable  = linkedBrokerParcelable.getLinkedLogin();
-            if (!linkedBrokers.contains(linkedBrokerParcelable)) {
-                if (linkedBrokerData.isLinkActivationPending){
-                    linkedBrokerParcelable.setAccountLinkDelayedError();
-                }
-                linkedBrokerCache.cache(linkedBrokerParcelable);
-                linkedBrokers.add(linkedBrokerParcelable);
-                keystoreService.saveLinkedLogin(linkedLoginParcelable, linkedLoginParcelable.label);
-            }
-        }
-
-        // Remove non existing linkedBrokers
+        // Remove existing linkedBrokers
         for (TradeItLinkedBrokerParcelable linkedBroker: new ArrayList<>(linkedBrokers)) {
-            if (!linkedBrokerDataList.contains(new TradeItLinkedBrokerData(linkedBroker.getLinkedLogin()))) {
                 linkedBrokerCache.removeFromCache(linkedBroker);
                 linkedBrokers.remove(linkedBroker);
                 keystoreService.deleteLinkedLogin(linkedBroker.getLinkedLogin());
+        }
```

Comment:
@jsom brought up a good point that we should only update accounts if they actually change because Yahoo will be calling this a lot (every 30 seconds) and writing to local storage for iOS/Android is slow.